### PR TITLE
[build] Pass list of .v files to coqdep using a file.

### PR DESCRIPTION
**Pull request by ejgallego**
This should help with Windows problems "command line argument limit".

Alternative to #14240

### CODE REVIEW

File: Makefile.vofiles [hunk 1]
```diff
@@ -28,6 +28,14 @@ VFILES := $(THEORIESV) $(CONTRIBV)
 # Build directory is needed to have the glob files
 BUILD_VFILES=$(addprefix $(VO_OUT_DIR),$(VFILES))
 
+BUILD_VFILES_FILE=$(BUILD_OUT_DIR).all_v_files
+
+# This has to be kept in sync with above, used in the windows build to
+# avoid coqdep maximum number arguments problems. Paths in file have
+# to be relative to the location of the coqdep input file.
+$(BUILD_VFILES_FILE): $(BUILD_VFILES)
```
Comment (SkySkimmer) [inline Makefile.vofiles:36]: The dep is on VFILES not BUILD_VFILES

File: Makefile.vofiles [hunk 2]
```diff
@@ -28,6 +28,14 @@ VFILES := $(THEORIESV) $(CONTRIBV)
...
+	$(HIDE)find theories user-contrib -type f -name "*.v" -fprintf "$@" "lib/coq/%p\n"
```
Comment (SkySkimmer) [inline Makefile.vofiles:37]: osx doesn't have printf, that's why I sed

File: Makefile.vofiles (hunk 1 above)
Comment (ejgallego) [reply Makefile.vofiles:36]: You are right, I was thinking of running coqdep on the copied sources as they should include auto-generated .v files if we ever had them; let me think about this one.

File: Makefile.vofiles (hunk 2 above)
Comment (ejgallego) [reply Makefile.vofiles:40]: Damn, while writing the patch I thought of checking that but I was lazy. But the situation seems a f***-up, all of `find` `sed` `xargs` and `stat` are different in OSX so I've opted for using a specific OSX command.

**Comment by SkySkimmer** [comment]
IMO you're wasting your time when you could just merge my PR

I definitely do not accept using a makefile `if` for this

**Comment by ejgallego** [comment]
> IMO you're wasting your time when you could just merge my PR

I don't understand what the code you have in that PR does, so that worries me.

**Comment by SkySkimmer** [comment]
There's nothing complicated about it, what confuses you?

**Review by SkySkimmer — CHANGES_REQUESTED** [review]
1 file is simpler than 2, let's just use my pr

**Comment by gares** [comment]
The other PR seems simpler to me, and also works.

**Comment by ejgallego** [comment]
Ok, let's close this one then; however the v file setup needs to be further tweaked.

**Pull request closed without merging**